packages/next-swc/crates/core/src/server_actions.rs

```diff
@@ -670,16 +534,22 @@ impl<C: Comments> VisitMut for ServerActions<C> {
                     .emit();
             });
         } else if !self.in_action_file {
-            let (_, maybe_new_fn) = self.add_action_annotations_and_maybe_hoist(
-                &f.ident,
-                Some(&mut f.function),
-                None,
-                false,
-            );
+            self.maybe_hoist_and_create_proxy(&f.ident, Some(&mut f.function), None);
 
-            if let Some(new_fn) = maybe_new_fn {
-                f.function = new_fn;
-            }
+            // Make the original function declaration empty, as we have hoisted it already.
+            f.function = Box::new(Function {
+                params: vec![],
+                decorators: vec![],
+                span: DUMMY_SP,
+                body: Some(BlockStmt {
+                    span: DUMMY_SP,
+                    stmts: vec![],
+                }),
+                is_generator: false,
+                is_async: false,
```

Comment:
Is this correct?

Reply:
Yeah this function will become an no-op, and we're setting its body to be empty so it doesn't matter.